_posts/2025-02-11-announcing-ballerina-2201.11.0-swan-lake-update-11.md

```diff
@@ -0,0 +1,162 @@
+---
+layout: blog-post
+title: Announcing Ballerina Swan Lake Update 11 (2201.11.0)
+author: Waruna Jayaweera
+published-date: 11 Feb 2025
+status: Published
+socialmediaimage: ballerina-generic-social-media-image-2023.png
+permalink: /posts/2025-02-11-announcing-ballerina-2201.11.0-swan-lake-update-11/
+---
+
+<style>.cBlogContent p{white-space: break-spaces !important;}</style>
+
+The Ballerina team is excited to announce the eleventh update release of Ballerina Swan Lake, Ballerina Swan Lake Update 11 (2201.11.0). This update adds support for Java 21, along with performance improvements and new features for the Ballerina library and developer tools.
+## Runtime
+### Java 21 Support
+The Ballerina runtime now supports Java 21, the latest long-term support (LTS) release of the Java platform. Ballerina's existing concurrency model has been implemented with Project Loom virtual threads and the newest concurrency features in Java 21.
+
+Ballerina models concurrency using the concept of "Strand". A new strand is created via a named worker, the 'start' action, or any entry point execution, such as the main function or a resource method. During execution, a strand can yield—for example, while waiting for a worker result or a network call—allowing the underlying platform thread to be shared with other strands.
+
+Java threads were mapped to platform threads, making it costly to create a Java thread for each strand. Therefore, strands could not be directly mapped to Java threads. Initially, Project Loom was in its early stages as well. So, Ballerina's own scheduler was implemented to manage strands, similar to Java 21 virtual threads. The Ballerina scheduler assigns a strand to an available Java thread from the Ballerina thread pool and manages yield points to prevent Java threads from being blocked.
```

Comment:
What are platform threads here? 

We need to establish that strands are lightweight threads similar to Go routines. Java did not have a green thread concept in earlier versions. Therefore, we implemented a user-space scheduler for Ballerina strands.

Reply:
I have refactored the content. Please check.